src/funnels/automation/VaultMinter.sol

```diff
@@ -0,0 +1,118 @@
+// SPDX-FileCopyrightText: © 2023 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.16;
+
+interface AllocatorVaultLike {
+    function draw(uint256) external;
+    function wipe(uint256) external;
+}
+
+contract VaultMinter {
+    mapping (address => uint256) public wards; // Admins
+    mapping (address => uint256) public buds;  // Whitelisted keepers
+    MinterConfig public config;                // Configuration for keepers
+
+    address public immutable vault; // The address of the vault contract
+
+    struct MinterConfig {
+        int64    num; // The remaining number of times that a mint or burn can be executed by keepers (> 0: mint, < 0: burn)
+        uint32   hop; // Cooldown period it has to wait between each action
+        uint32   zzz; // Timestamp of the last action
+        uint128  lot; // The amount to mint or burn every hop
+    }
+
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event Kiss(address indexed usr);
+    event Diss(address indexed usr);
+    event SetConfig(int64 num, uint32 hop, uint128 lot);
+    event Mint(uint128 lot);
+    event Burn(uint128 lot);
```

Comment:
Should we change these to Draw and Wipe now?

Reply:
Fixed.